src/piece_picker.cpp

```diff
@@ -727,12 +727,11 @@ namespace libtorrent {
 		// and also the number of pieces that have more than that.
 		int integer_part = 0;
 		int fraction_part = 0;
-		for (std::vector<piece_pos>::const_iterator i = m_piece_map.begin()
-			, end(m_piece_map.end()); i != end; ++i)
+		for (auto i : m_piece_map)
```

Comment:
this should be `auto const& i`, shouldn't it?

Reply:
No idea. I also ran this through clang’s -Wrange-loop-analysis . Didn’t complain.

Reply:
your change will make a copy of the `piece_pos` object. The previous code didn't